File: pset1/cash/cash.c
//Implement a program that calculates the minimum number of coins required to give a user change.


#include <stdio.h>
#include <cs50.h>
#include <math.h>


int main(void)
{
    float change;
    do //verifying user input is non negative
    {
        change = get_float("Cash: ");
    }
    while (change < 0);


    int casted = round(change * 100);   //changing float to int
    int coins = 0;

    while (casted > 0)
    {
        coins ++;          //increment coins returned
        if (casted >= 25)
        {
            casted -= 25;
        }
        else if (casted >= 10)
        {
            casted -= 10;
        }
        else if (casted >= 5)
        {
            casted -= 5;
        }
        else
        {
            casted -= 1;
        }
    }
    printf("%i\n", coins);         //return coins
}

File: pset6/cash/cash.py
# Implement a program that calculates the minimum number of coins required to give a user change.

cash = -1
while cash < 0:
    try:
        cash = float(input("Change owed: "))
    except ValueError:
        cash = -1

coins_available = [0.25, 0.10, 0.05, 0.01]
coins = 0
while cash > 0:
    for coin in coins_available:
        if coin <= cash:
            coins+=1
            cash = round(cash - coin, 2)
            break

print(coins)